--- conf.d/default.conf.orig	2016-11-17 20:32:10.807414912 +0800
+++ conf.d/default.conf.mod	2016-11-17 20:33:51.442673362 +0800
@@ -10,6 +10,17 @@
         index  index.html index.htm;
     }
 
+    location /teamcity/ {
+        proxy_pass http://${TEAMCITY_SERVER_IP}:8111/teamcity/;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header Host $host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+
+        proxy_http_version 1.1;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection "upgrade";
+    }
+
     #error_page  404              /404.html;
 
     # redirect server error pages to the static page /50x.html
